packages/grid/x-data-grid-pro/src/internals/hooks/features/rowGrouping/createGroupingColDef.tsx

```diff
@@ -198,15 +198,15 @@ export const createGroupingColDefForOneGroupingCriteria = ({
           return <GridGroupingColumnLeafCell {...leafParams} />;
         }
 
-        return null;
+        return '';
```

Comment:
When `renderCell` returns `null`, the grid fallbacks on the formatted value (I want to change that on v6).
So if a user sets a value formatter on a grouping column with `rowGroupingColumnMode="multiple"`, it renders also on the cells of the other grouping criterias.

Returning `''` here avoid this fallback and make sure that the value returned by `renderCell` is always the one displayed.